addons/stock_maintenance/models/maintenance.py

```diff
@@ -1,33 +1,41 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
 from odoo import api, fields, models
 
 
 class MaintenanceEquipment(models.Model):
     _inherit = 'maintenance.equipment'
 
     location_id = fields.Many2one('stock.location', 'Location', domain="[('usage', '=', 'internal')]")
-    match_serial = fields.Boolean(compute='_compute_match_serial')
+    matching_serials = fields.Many2many('stock.lot', compute='_compute_matching_serials')
 
     @api.depends('serial_no')
-    def _compute_match_serial(self):
-        if not self.env['stock.lot'].has_access('read') or not self.env.user.has_group('stock.group_production_lot'):
-            self.match_serial = False
+    def _compute_matching_serials(self):
+        self.matching_serials = False
+        if not (self.env.user.has_group('stock.group_stock_user') and
+                self.env.user.has_group('stock.group_production_lot')):
             return
-        matched_serial_data = self.env['stock.lot']._read_group(
-            [('name', 'in', self.mapped('serial_no'))],
-            ['name'],
-            ['__count'],
-        )
-        matched_serial_count = dict(matched_serial_data)
+        serials = self.env['stock.lot'].search_fetch([], ['name'])
```

Comment:
No bueno :-1: , there can potentially be a lot of existing stock.lot records and we don't want to risk an out of memory error loading all of them at the same time. What would be cleaner is to leave this compute as is and then make `action_open_matched_serial` do a `search` for `stock.lot` records that match the name and then do the form or list view logic based on the returned values.